### ISSUE TITLE
Cop idea: prefer each over map when return value is not used

### ISSUE THREAD

**Issue by okuramasafumi** [rubocop-hq/rubocop#8270]
## Is your feature request related to a problem? Please describe.

I found a code like this:

```ruby
Model.all.map { do_something_as_side_effect }
```

I think everyone will agree that we should use `each` instead of `map` in this case, since `map` is mainly used for its return value but here it's not used.

This sort of thing can be automated by RuboCop!

## Describe the solution you'd like

When a method named `map` is used and the return value is not used, the cop warns.
It can support auto-correct to replace `map` with `each`.

## Describe alternatives you've considered

Actually, there are some more methods from `Enumerable` module that we use for their return values, such as `select`.
We might let users specify method names to replace with `each`

## Additional context
 
When a user defines their own method named `map`, this cop might produce false positive. However, when it's named as `map`, it clearly shows that it's expected to return value and we care it.

**Comment by fatkodima** [comment]
Duplicate https://github.com/rubocop-hq/rubocop/issues/1052 ?

**Comment by okuramasafumi** [comment]
Thanks I didn't notice this old issue.

**Comment by koic** [comment]
@fatkodima Good catch!
This issue has been closed in the past. Opening as a PR would be welcome. See #1052 for more information. Thank you.
